apple/Sources/FerrostarCore/NavigationState.swift

```diff
@@ -47,11 +47,29 @@ public struct NavigationState: Hashable {
         return remainingSteps
     }
 
+    /// The current geometry segment's annotations in a JSON string.
+    ///
+    /// A segment is the line between two coordinates on the geometry.
     public var currentAnnotationJSON: String? {
         guard case let .navigating(_, _, _, _, _, _, _, _, annotationJson: annotationJson) = tripState else {
             return nil
         }
 
         return annotationJson
     }
+
+    /// The current geometry segment's annotations.
+    ///
+    /// A segment is the line between two coordinates on the geometry.
+    ///
+    /// - Parameter type: The type to decode the annotation json string to.
+    /// - Returns: The decoded type.
+    public func currentAnnotation<T: Decodable>(as type: T.Type) throws -> T? {
```

Comment:
this method throwing and return optional at the same time which makes it harder to use, probably throwing error in case of nil date would make more sense.

Reply:
I agree it's a bit harder to use than some of our functions, but there's a reason:

1. `nil` means there is no current attribute/no attribute data on the route.
2. throw occurs when there's an actual issue with the route JSON and the `JSONDecoder` fails. This is probably something a developer or logger would want to understand. It might occur for example if their API format is different than their annotation model. A developer could still `try?` if they didn't care about malformed JSON.

I'll add a `- Throws:` to the docstring to explain that. Thoughts on an example excerpt as well to show how it could be used in different cases? It should be easy enough to handle gracefully with some examples.

Reply:
This is gone in favor of a bespoke `AnnotationPublisher` on the core 👍